src/lib.rs

```diff
@@ -413,15 +587,16 @@ mod tests {
 
         // Individual location line, e.g:
         //  "#1: Register #R0, size: 8"
-        //  Skip for now
+        let mut locs = Vec::with_capacity(num_locs as usize);
```

Comment:
Is this `let locs = lines.next().map(|x| parse_loc(x.unwrap()))` in disguise?

Reply:
Not quite, although that would be prettier if it was. We only want to iterate over a subset of lines. `num_locs` <= `lines`.

`let locs = lines.next().map(|x| parse_loc(x.unwrap()))` would iterate beyond the locations and into another stackmap record's output.

Reply:
`let locs = lines.next().take(num_locs).map(|x| parse_loc(x.unwrap()))` maybe?